lib/collections.js

```diff
@@ -26,9 +29,21 @@ var normalizedWords = [];
 
 // The words stored as keys of a dictionary
 var uniqueWordsAsKeys = {};
+for (var i = 0; i < normalizedWords.length; i++) {
+  var key = normalizedWords[i];
+  if (uniqueWordsAsKeys[key] === undefined) {
+    uniqueWordsAsKeys[key] = true;
+  } else {
+    uniqueWordsAsKeys[key] = false;
+  }
+  console.log("uniqueWordsAsKeys now has keys..." uniqueWordsAsKeys.keys.toString());
+}
 
 // The keys pulled out into a list
 var uniqueWords = [];
+for (var key in uniqueWordsAsKeys) {
+  uniqueWords[uniqueWords.length] = key;
+}
```

Comment:
Huh! I've never seen that approach (`uniqueWords[uniqueWords.length]`) before, but it looks like it JavaScript allows it!

Not all languages will allow you to define new array indices on the fly in this way, so an approach I'd recommend instead is to use `.push`, which will always put something onto the end of an array.